### PR TITLE
deploy button checks refactor, upgrade gas limit calc fix

### PR DESCRIPTION
**Pull request by poocart**
Includes:
1. Fix for Smart Wallet account creation during upgrade flow.
2. Moved displaying upgrade messages and deploy button to single component.

### CODE REVIEW

File: src/actions/smartWalletActions.js [hunk 1]
```diff
@@ -1282,19 +1282,35 @@ export const getAssetTransferGasLimitsAction = () => {
...
+      await smartWalletService.sdk.initialize().catch(() => null);
+    }
+    const smartAccounts = await smartWalletService.getAccounts().catch(() => null);
+    if (!smartAccounts || !smartAccounts.length) {
+      /**
+       * let's create account, it will be later fetched or new will be created if re-init happens
+       * we need smart wallet account address for precise gas limit calculation
+       */
+      const tempAccount = await smartWalletService.sdk.createAccount().catch(() => null);
+      if (!tempAccount) {
```
Comment (grilix) [inline src/actions/smartWalletActions.js]: Prefer positive first on if statements.

File: src/screens/Assets/WalletView.js [hunk 2]
```diff
@@ -48,14 +48,15 @@ import {
...
+import { SMART_WALLET_UPGRADE_STATUSES } from 'constants/smartWalletConstants';
 
 import { activeAccountSelector } from 'selectors';
 import { accountBalancesSelector } from 'selectors/balances';
 import { accountCollectiblesSelector } from 'selectors/collectibles';
 import Spinner from 'components/Spinner';
 import Separator from 'components/Separator';
 import EmptyStateParagraph from 'components/EmptyState/EmptyStateParagraph';
-import { DeploymentView, getDeployErrorMessage } from 'components/DeploymentView';
+import DeploymentView, { getDeployErrorMessage } from 'components/DeploymentView';
```
Comment (alazarevski) [inline src/screens/Assets/WalletView.js]: not sure if utils would be a better place to store the functions

File: src/actions/smartWalletActions.js (hunk 1 above)
Comment (poocart) [reply src/actions/smartWalletActions.js]: This checks if no `tempAccount` and shows toast and blocking from further code, I guess in this case it's fine.

File: src/screens/Assets/WalletView.js (hunk 2 above)
Comment (poocart) [reply src/screens/Assets/WalletView.js]: moved

File: src/actions/smartWalletActions.js (hunk 1 above)
Comment (grilix) [reply src/actions/smartWalletActions.js]: Damn, sorry, I got confused with the `else` from the other `if` :man_facepalming:

**Pull request merged**